agent/indy_catalyst_agent/messaging/credentials/manager.py

```diff
@@ -49,37 +51,90 @@ def context(self) -> RequestContext:
         """
         return self._context
 
-    async def create_offer(
-        self,
-        credential_definition_id,
-        connection_id,
-        auto_issue=None,
-        credential_values=None,
+    async def prepare_send(
+        self, credential_definition_id, connection_id, credential_values
     ):
         """
         Create an offer.
 
         Args:
             credential_definition_id: Credential definition id for offer
             connection_id: Connection to create offer for
-            auto_issue: Should requests for this offer automatically be handled to
-                issue a credential
             credential_values: The credential values to use if auto_issue is enabled
 
         Returns:
-            A tuple (credential_exchange, credential_offer_message)
+            A tuple (
+                credential_exchange,
+                credential_offer_message
+            )
 
         """
 
-        # If auto_issue is False, ignore credential_values
-        if not auto_issue:
-            credential_values = None
+        cache: BaseCache = await self._context.inject(BaseCache)
+
+        # This cache is populated in credential_request_handler.py
+        # Do we have a source (parent) credential exchange for which
+        # we can re-use the credential request/offer?
+        source_credential_exchange_id = await cache.get(credential_definition_id)
```

Comment:
I'm not sure credential_definition_id is unique enough since you may be issuing to multiple targets. Maybe credential_definition_id + connection_id?

Reply:
Good catch